gateway-service-impl/src/main/java/org/hypertrace/gateway/service/EntityTypesProvider.java

```diff
@@ -0,0 +1,55 @@
+package org.hypertrace.gateway.service;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.hypertrace.entity.type.service.client.EntityTypeServiceClient;
+import org.hypertrace.entity.type.service.v1.EntityType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EntityTypesProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(EntityTypesProvider.class);
+  private static final int DEFAULT_CACHE_SIZE = 1024;
+  private static final int DEFAULT_EXPIRE_DURATION_MIN = 30; // 30 minutes
+
+  // TenantId to Entity types cache
+  private final LoadingCache<String, List<String>> tenantIdToEntityTypesCache;
+
+  public EntityTypesProvider(EntityTypeServiceClient entityTypeServiceClient) {
+    CacheLoader<String, List<String>> cacheLoader =
+        new CacheLoader<>() {
+          @Override
+          public List<String> load(@NonNull String tenantId) {
+            return entityTypeServiceClient.getAllEntityTypes(tenantId).stream()
+                .map(EntityType::getName)
+                .collect(Collectors.toList());
+          }
+        };
+
+    tenantIdToEntityTypesCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(DEFAULT_CACHE_SIZE)
+            .expireAfterWrite(DEFAULT_EXPIRE_DURATION_MIN, TimeUnit.MINUTES)
+            .build(cacheLoader);
+  }
+
+  public List<String> getEntityTypes(String tenantId) {
+    if (StringUtils.isBlank(tenantId)) {
+      LOG.error("Tenant id not found while trying to fetch entity types");
```

Comment:
shouldnt we return or throw an error here? whats the behaviour in this case? do we have any default case?

Reply:
Since we are relying on the enum based too, I had not thrown error and just logged it, so old flow can still support the queries.
If we want to remove enum based entity types altogether, we can throw the error here